sdk/cosmos/azure-cosmos/src/main/java/com/azure/cosmos/implementation/throughputControl/ThroughputControlStore.java

```diff
@@ -104,27 +104,30 @@ public ThroughputControlStore(
         this.globalEndpointManager = globalEndpointManager;
         this.groupMapByContainer = new ConcurrentHashMap<>();
         this.partitionKeyRangeCache = partitionKeyRangeCache;
+
+        this.cancellationTokenSource = new LinkedCancellationTokenSource();
+        this.cancellationTokenMap = new ConcurrentHashMap<>();
     }
 
-    public void enableThroughputControlGroup(ThroughputControlGroup group) {
+    public void enableThroughputControlGroup(ThroughputControlGroupInternal group) {
         checkNotNull(group, "Throughput control group cannot be null");
 
-        String collectionLink = Utils.trimBeginningAndEndingSlashes(
-            BridgeInternal.extractContainerSelfLink(
-                BridgeInternal.getTargetContainerFromThroughputControlGroup(group)));
+        String collectionLink = Utils.trimBeginningAndEndingSlashes(BridgeInternal.extractContainerSelfLink(group.getTargetContainer()));
         this.groupMapByContainer.compute(collectionLink, (key, groupSet) -> {
             if (groupSet == null) {
                 groupSet = new HashSet<>();
```

Comment:
shouldn't this be `ConcurentHashSet` to ensure thread-safety?

Reply:
yup, changed to use concurrentHashSet